lib/miq_automation_engine/models/miq_ae_domain.rb

```diff
@@ -31,6 +31,22 @@ def default_priority
     self.priority = MiqAeDomain.highest_priority(tenant) + 1 unless priority
   end
 
+  def version
+    system = ae_namespaces.detect { |ns| ns.name.downcase == 'system' }
+    about = system.try(:ae_classes).try(:detect) { |cls| cls.name.downcase == 'about' }
+    version_field = about.try(:ae_fields).try(:detect) { |fld| fld.name == 'version' }
+    version_field.try(:default_value)
+  end
+
+  def available_version
+    fname = about_file_name.to_s
+    return nil unless File.exist?(fname)
```

Comment:
The `.to_s` call seems odd to me.  Why not check for `nil` on the follow line and then you do not need the to_s at all.
`return nil if fname.nil? || !File.exist?(fname)`